Bug-Ubuntu: https://bugs.launchpad.net/bugs/1986852
Origin: upstream, https://git.savannah.nongnu.org/gitweb/?p=dmidecode.git;a=commit;h=4304d3aca801534ae0e4ebab5e4b4cef0c866315
From: Prabhakar pujeri <prabhakar.pujeri@gmail.com>
Date: Mon, 19 Oct 2020 15:04:30 +0200
Subject: [PATCH] Add new slot characteristics2 from SMBIOS spec 3.4.0

* The bit to indicate PCIe surprise removal support.
* The bit to indicate Flexbus CXL 1.0 version.
* The bit to indicate Flexbus CXL 2.0 version.
---
 dmidecode.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/dmidecode.c b/dmidecode.c
index eb5a5afcae72..1b7db4b62eed 100644
--- a/dmidecode.c
+++ b/dmidecode.c
@@ -2118,7 +2118,10 @@ static void dmi_slot_characteristics(const char *attr, u8 code1, u8 code2)
 		"PME signal is supported", /* 0 */
 		"Hot-plug devices are supported",
 		"SMBus signal is supported",
-		"PCIe slot bifurcation is supported" /* 3 */
+		"PCIe slot bifurcation is supported",
+		"Async/surprise removal is supported",
+		"Flexbus slot, CXL 1.0 capable",
+		"Flexbus slot, CXL 2.0 capable" /* 6 */
 	};
 
 	if (code1 & (1 << 0))
@@ -2133,7 +2136,7 @@ static void dmi_slot_characteristics(const char *attr, u8 code1, u8 code2)
 		for (i = 1; i <= 7; i++)
 			if (code1 & (1 << i))
 				pr_list_item("%s", characteristics1[i - 1]);
-		for (i = 0; i <= 3; i++)
+		for (i = 0; i <= 6; i++)
 			if (code2 & (1 << i))
 				pr_list_item("%s", characteristics2[i]);
 		pr_list_end();
-- 
2.34.1

